### PR TITLE
release 0.3.0

### PR DESCRIPTION
**Pull request by zhongqiangczq**

### CODE REVIEW

File: docs/community/news.md [hunk 1]
```diff
@@ -18,6 +18,7 @@ license: |
 
 | Date            | Title                                                   | Brief                                                                                           |
 |-----------------|---------------------------------------------------------|-------------------------------------------------------------------------------------------------|
+| 2023 July 29    | Release 0.3.0-incubating
```
Comment (waitinfuture) [inline docs/community/news.md]: Just curious, why July 29?

Comment (zhongqiangczq) [reply docs/community/news.md]: it's my fault. it should be July 27.

File: .github/workflows/site.yaml [hunk 2]
```diff
@@ -59,16 +59,16 @@ jobs:
           git add .
           git commit -m 'docs/main'
       - run: |
-          wget https://github.com/apache/incubator-celeborn/archive/refs/tags/v0.2.1-incubating.tar.gz
-          tar -xzf v0.2.1-incubating.tar.gz
-          cd incubator-celeborn-0.2.1-incubating
+          wget https://github.com/apache/incubator-celeborn/archive/refs/tags/v0.3.0-incubating.tar.gz
```
Comment (waitinfuture) [inline .github/workflows/site.yaml:73]: Shouldn't we retain 0.2.1 here?

**Comment by waitinfuture** [comment]
cc @pan3793

File: .github/workflows/site.yaml (hunk 2 above)
Comment (pan3793) [reply .github/workflows/site.yaml:73]: we should retain all history-versioned docs

Comment (zhongqiangczq) [reply .github/workflows/site.yaml:73]: i think that this code should be update to  produce history docs.
image

**Comment by zhongqiangczq** [comment]
>  @pan3793， how to change the git tag to 0.3.0. 

image

**Comment by pan3793** [comment]
@zhongqiangczq Create a new GitHub release using tag `v0.3.0-incubating`, then it will be updated automatically.

File: docs/community/release_notes/release_note_0.3.0.md [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+- Performance improvements and bug fixs of native Spark columnar shuffle
+
+- More bug fixes and usability improvements of K8S deployment
+
+- More code reflectors and code refines
+
+- More configuration default values are changed and new confs are added
+
+### Spark/Flink
+- FLINK: Implement Flink 1.14 plugin [CELEBORN-235]
```
Comment (pan3793) [inline docs/community/release_notes/release_note_0.3.0.md]: How about moving the ticket ahead? (No strong opinion)

```suggestion
- [CELEBORN-235] FLINK: Implement Flink 1.14 plugin
```

**Comment by zhongqiangczq** [comment]
> v0.3.0-incubating
but I have already created v0.3.0-incubating, it's still v0.2.1-incubating
image

File: docs/community/release_notes/release_note_0.3.0.md (hunk 3 above)
Comment (zhongqiangczq) [reply docs/community/release_notes/release_note_0.3.0.md]: ok. i move them later.

**Comment by pan3793** [comment]
>  Create a new **GitHub release** using tag ...

image

File: .github/workflows/site.yaml (hunk 2 above)
Comment (pan3793) [reply .github/workflows/site.yaml:73]: This script needs to improve, e.g. extract the duplicated code. Ideally, we only need to update the version list when a new release happens.

Anyway, the optimization is not a blocker of this PR, copy-paste is fine, and we can optimize it later.

File: docs/community/release_notes/release_note_0.3.0.md [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+  - navigation
+
+license: |
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+  http://www.apache.org/licenses/LICENSE-2.0
```
Comment (pan3793) [inline docs/community/release_notes/release_note_0.3.0.md]: this line format is broken (actually many files are broken ...)

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
@zhongqiangczq thank you for driving the release of Celeborn 0.3.0

File: docs/community/release_notes/release_note_0.3.0.md (hunk 4 above)
Comment (zhongqiangczq) [reply docs/community/release_notes/release_note_0.3.0.md]: ok. this line prefix shold fill some spaces.

**Pull request merged**